auth/tests/acc/attestation.test.ts

```diff
@@ -1,24 +1,183 @@
+const functionCommand = new GetFunctionCommand({
+  FunctionName: testConfig.authProxyFunctionName,
+});
+
+const iamClient = new IAMClient({ region: "eu-west-2" });
+
+const iamCommandForAuthProxyLambdaRole = new GetRoleCommand({
+  RoleName: testConfig.authProxyFunctionIAMRoleName,
+});
+
+const iamCommandGetRolePolicy = new GetRolePolicyCommand({
+  RoleName: testConfig.authProxyFunctionIAMRoleName,
+  PolicyName: testConfig.authProxyFunctionIAMRolePolicyName,
+});
 
 const REGION = process.env.CFN_AWS_REGION || "eu-west-2";
 
 describe("attestation", async () => {
   describe("attestation proxy is a confidential client", () => {
     const secretsManagerClient = new SecretsManagerClient({ region: REGION });
-  
+
     it("retrieves shared signal credentials from Secrets Manager", async () => {
       const secretId = testConfig.cognitoSecretName;
-  
-      const getSecretCommand = new GetSecretValueCommand({ SecretId: secretId });
+
+      const getSecretCommand = new GetSecretValueCommand({
+        SecretId: secretId,
+      });
       const secretResponse = await secretsManagerClient.send(getSecretCommand);
-  
+
       expect(secretResponse.SecretString).toBeDefined();
-  
+
       const secretData = JSON.parse(secretResponse.SecretString as string);
       expect(secretData).toHaveProperty("client_secret");
     });
   });
-});
\ No newline at end of file
+
+  describe("deployed lambda configuration", async () => {
+    const response = await lambdaClient.send(functionCommand);
+    const attestationLambda = response.Configuration!;
+
+    it("has the correct timeout", () => {
+      assert.equal(attestationLambda.Timeout, 60);
+    });
+
+    it("has the correct memory size", () => {
+      assert.equal(attestationLambda.MemorySize, 128);
+    });
+
+    it("has the correct tracing configuration", () => {
+      assert.equal(attestationLambda.TracingConfig?.Mode, "PassThrough");
+    });
+
+    it("has the correct runtime", () => {
+      assert.equal(attestationLambda.Runtime, "nodejs22.x");
+    });
+
+    it("has the correct handler", () => {
+      assert.equal(attestationLambda.Handler, "app.lambdaHandler");
+    });
+
+    it("has the correct package type", () => {
+      assert.equal(attestationLambda.PackageType, "Zip");
+    });
+
+    it("has the correct state", () => {
+      assert.equal(attestationLambda.State, "Active");
+    });
+
+    it("has the correct ephemeral storage size", () => {
+      assert.equal(attestationLambda.EphemeralStorage?.Size, 512);
+    });
+
+    it("has the correct architecture", () => {
+      assert.deepEqual(attestationLambda.Architectures, ["x86_64"]);
+    });
+
+    it("has the correct logging configuration", () => {
+      assert.equal(attestationLambda.LoggingConfig?.LogFormat, "Text");
+      assert.equal(
+        attestationLambda.LoggingConfig?.LogGroup,
+        `/aws/lambda/${testConfig.authProxyFunctionName}`
+      );
+    });
```

Comment:
Should we have this as a test for all lambda functions? Else it might cause a bit of bloat if we have to assert this for each new lambda?